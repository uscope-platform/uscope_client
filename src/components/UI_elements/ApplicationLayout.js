import styled,{css} from "styled-components";

const ApplicationLayout = styled.div`
    display: grid;
    grid-template-columns: 200px auto 300px;
    grid-auto-rows: auto;
    height: 100vh;
    ${props => !props.sidebarNeeded && css`
        grid-template-columns: 200px  auto;
    `}
 
`

const theme = {
    base: "#08081A",
    level_1:"#162447",
    level_2:"#1f4068",
    accents:"#e43f5a"
};

export default ApplicationLayout
