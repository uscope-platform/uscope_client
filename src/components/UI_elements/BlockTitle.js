import styled from "styled-components";

const BlockTitle = styled.h1`
font-size: 24px;
margin: 0 auto;
`

export default BlockTitle;
