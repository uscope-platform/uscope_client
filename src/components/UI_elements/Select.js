// Copyright 2021 University of Nottingham Ningbo China
// Author: Filippo Savi <filssavi@gmail.com>
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import styled from 'styled-components';
import {Label} from "./Label";
import React from "react";



export const Select = styled.select`
  width: fit-content;
  height: 2rem;
  border-radius: 5px;
  min-width: 4em;
  option {
    display: flex;
    justify-content: center;
    min-height: 20px;
  }
`;


const SelectWrapper = styled.div`
    display: grid;
    grid-template-columns: repeat(2, auto);
    grid-gap: 0.3rem;
    justify-content: space-between;
    align-items: start;
`

export let  SelectField = props =>{
    return(
        <SelectWrapper>
            <Label>{props.label}</Label>
            <Select name={props.name} id={props.name} defaultValue={props.defaultValue} onChange={props.onChange}>
                <option value={props.name} hidden>{props.placeholder}</option>
                {
                    props.options.map((name,i) => (
                        <option key={i} >{name}</option>
                    ))
                }
            </Select>
        </SelectWrapper>
    );
};

