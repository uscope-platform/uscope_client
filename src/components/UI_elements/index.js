// Copyright 2021 University of Nottingham Ningbo China
// Author: Filippo Savi <filssavi@gmail.com>
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

export {StyledScrollbar} from "./StyledScrollbar"
export {Select, SelectField} from "./Select"
export {Label} from "./Label"
export {InputField} from "./InputField"
export {Image} from "./Image"
export {FileChoice} from "./FileChoice"
export {ColorTheme} from "./ColorTheme"
export {Checkbox} from "./checkbox"
export {BlockTitle} from "./BlockTitle"
export {Button} from "./Button"
export {MultiSelect} from "./MultiSelect"
export {SelectableListItem} from './SelectableListItem'
export {SelectableListLayout} from './SelectableListLayout'
export {SelectableList} from './SelectableList'

export {ApplicationLayout} from "./Layouts/ApplicationLayout"
export {BlockLayout} from "./Layouts/BlockLayout"
export {FormLayout} from "./Layouts/FormLayout"
export {ManagerLayout} from "./Layouts/ManagerLayout"
export {ManagerButtonsLayout} from "./Layouts/ManagerLayout"
export {SidebarBlockLayout} from "./Layouts/SidebarBlockLayout"
export {SidebarBlockTitleLayout} from "./Layouts/SidebarBlockTitleLayout"
export {SidebarCollapsableNameLayout} from "./Layouts/SidebarCollapsableNameLayout"
export {SidebarCollapsableContentLayout} from "./Layouts/SidebarCollapsableContentLayout"
export {SidebarContentLayout} from "./Layouts/SidebarContentLayout"
export {SidebarLayout} from "./Layouts/SidebarLayout"

export {ApplicationSoftCoreProperties} from "./SidebarComponents/ApplicationSoftCoreProperties"
export {ApplicationMiscFieldProperties} from "./SidebarComponents/ApplicationMiscFieldProperties"
export {ApplicationPeripheralProperties} from "./SidebarComponents/ApplicationPeripheralProperties"
export {InitialRegisterValue} from "./SidebarComponents/InitialRegisterValueProperties"
export {MacroProperties} from "./SidebarComponents/MacroProperties"
export {ParameterProperties} from "./SidebarComponents/ParameterProperties"
export {PlotChannelProperties} from "./SidebarComponents/PlotChannelProperties"
export {PlotChannelGroupProperties} from  "./SidebarComponents/PlotChannelGroupProperties"
export {RegisterProperties} from "./SidebarComponents/RegisterProperties"

export {SimpleContent} from './panels/SimpleContent'
export {TabbedContent} from './panels/TabbedContent'
export {UIPanel} from './panels/UIPanel'