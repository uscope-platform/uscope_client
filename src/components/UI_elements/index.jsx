// Copyright 2021 University of Nottingham Ningbo China
// Author: Filippo Savi <filssavi@gmail.com>
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

export {StyledScrollbar} from "./StyledScrollbar"
export {SelectField} from "./Select"
export {Label} from "./Label"
export {InputField} from "./InputField"
export {Image} from "./Image"
export {FileChoice} from "./FileChoice"
export {ColorTheme} from "./ColorTheme"
export {Checkbox} from "./checkbox"
export {BlockTitle} from "./BlockTitle"
export {Button} from "./Button"
export {MultiSelect} from "./MultiSelect"
export {SelectableListItem} from './SelectableListItem'
export {SelectableListLayout} from './SelectableListLayout'
export {SelectableList} from './SelectableList'
export {DeleteIcon} from './DeleteIcon'
export {ChipSelector} from './ChipSelector'
export {ListItem} from './ListItem'
export {TwoColumnSelector} from "./TwoColumnSelector"
export {Chip} from "./Chip"
export {Radio} from "./Radio"

export {FormLayout} from "./Layouts/FormLayout"
export {SidebarCollapsableNameLayout} from "./Layouts/SidebarCollapsableNameLayout"
export {SidebarCollapsableContentLayout} from "./Layouts/SidebarCollapsableContentLayout"

export {SoftCoreProperties} from "./SidebarComponents/Application/SoftCoreProperties"
export {FilterProperties} from "./SidebarComponents/Application/FiterProperties"
export {MiscFieldProperties} from "./SidebarComponents/Application/MiscFieldProperties"
export {PeripheralProperties} from "./SidebarComponents/Application/PeripheralProperties"
export {InitialRegisterValue} from "./SidebarComponents/Application/InitialRegisterValueProperties"
export {MacroProperties} from "./SidebarComponents/Application/MacroProperties"
export {ParameterProperties} from "./SidebarComponents/Application/ParameterProperties"
export {PlotChannelProperties} from "./SidebarComponents/Application/PlotChannelProperties"
export {PlotChannelGroupProperties} from "./SidebarComponents/Application/PlotChannelGroupProperties"
export {RegisterProperties} from "./SidebarComponents/RegisterProperties"


export {SimpleContent} from './panels/SimpleContent'
export {TabbedContent} from './panels/TabbedContent'
export {UIPanel} from './panels/UIPanel'
export {Card} from './panels/Card'
export {CardTitle} from './panels/CardTitle'
export {CardStack} from './panels/CardStack'
