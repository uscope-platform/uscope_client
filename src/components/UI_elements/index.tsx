// Copyright 2021 University of Nottingham Ningbo China
// Author: Filippo Savi <filssavi@gmail.com>
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

export {SelectField} from "./Select.js"
export {Label} from "./Label.js"
export {InputField} from "./InputField.js"
export {Image} from "./Image.js"
export {FileChoice} from "./FileChoice.js"
export {ColorTheme} from "./ColorTheme.js"
export {PlotConfigurations} from "./PlotConfigurations.js"
export {Checkbox} from "./checkbox.js"
export {BlockTitle} from "./BlockTitle.js"
export {Button} from "./Button.js"
export {DeleteButton} from "./DeleteButton.js"
export {MultiSelect} from "./MultiSelect.js"
export {SelectableListItem} from './SelectableListItem.js'
export {SelectableList} from './SelectableList.js'
export {ChipSelector} from './ChipSelector.js'
export {ListItem} from './ListItem.js'
export {TwoColumnSelector} from "./TwoColumnSelector.js"
export {Chip} from "./Chip.js"
export {RangedInputField} from "./RangedInputField.js"
export {TextEditor} from "./TextEditor.jsx"
export {SidebarBase} from "./Sidebar/SidebarBase.jsx"
export {SideToolbar} from "./Sidebar/SideToolbar.jsx"


export {InterfaceParameters} from "./InterfaceParameters.js"

export {FormLayout} from "./Layouts/FormLayout.js"
export {SidebarCollapsableNameLayout} from "./Layouts/SidebarCollapsableNameLayout.js"
export {SidebarCollapsableContentLayout} from "./Layouts/SidebarCollapsableContentLayout.js"

export {SimpleContent} from './panels/SimpleContent.js'
export {TabbedContent} from './panels/TabbedContent.js'
export {UIPanel} from './panels/UIPanel.js'
export {Card} from './panels/Card.js'
export {CardTitle} from './panels/CardTitle.js'
export {CardStack} from './panels/CardStack.js'
