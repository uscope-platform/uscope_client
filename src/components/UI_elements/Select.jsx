// Copyright 2021 University of Nottingham Ningbo China
// Author: Filippo Savi <filssavi@gmail.com>
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import styled from 'styled-components';
import {Label} from "./Label";
import React, {useCallback, useMemo} from "react";
import Select from "react-select";
import {ColorTheme} from "./ColorTheme";


const SelectWrapper = styled.div`
    display: grid;
    grid-template-columns: repeat(2, auto);
    grid-gap: 0.3rem;
    justify-content: space-between;
    align-items: start;
`


export let  SelectField = props =>{

    const color= props.color ? props.color : ColorTheme.background.select_background ;

    const style = useMemo(()=>{
        return {
            control:(provided,  { data, isDisabled, isFocused, isSelected }) => ({
                ...provided,
                minWidth:"7em",
                backgroundColor: color,
            }),

            menu: (provided,  { data, isDisabled, isFocused, isSelected }) => ({
                ...provided,
                width: "max-content",
                backgroundColor: color
            }),

            menuPortal: (provided,  { data, isDisabled, isFocused, isSelected }) => ({
                ...provided,
            }),

            option: (provided,  { data, isDisabled, isFocused, isSelected }) => ({
                ...provided,
                backgroundColor: isFocused?ColorTheme.background.transparent_accents:undefined
            }),

            singleValue:(provided,  { data, isDisabled, isFocused, isSelected }) => ({
                ...provided,
                color: ColorTheme.text
            })
        };
    }, [ColorTheme])

    const onSelectChange = useCallback((value, event) => props.onChange(value, event), [props.onChange]);

    return(
        <SelectWrapper>
            <Label htmlFor={props.name} >{props.label}</Label>
            <Select
                name={props.name}
                id={props.name}
                style={props.style}
                styles={style}
                defaultValue={props.defaultValue}
                value={props.value ? props.value: undefined}
                menuPortalTarget={document.body}
                onChange={onSelectChange}
                color={props.color ? props.color : ColorTheme.background.level_3}
                options={props.options}
            />
        </SelectWrapper>
    );
};

