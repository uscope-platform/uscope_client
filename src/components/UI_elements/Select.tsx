// Copyright 2021 University of Nottingham Ningbo China
// Author: Filippo Savi <filssavi@gmail.com>
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import {styled} from 'goober';
import {Label} from "./Label.js";
import React, {type JSX, useCallback, useMemo} from "react";
import Select, {type ActionMeta} from "react-select";
import {ColorTheme} from "./ColorTheme.js";
import type {DefaultOption} from "#interfaces/index.js";

const SelectWrapper = styled('div')`
    display: grid;
    grid-template-columns: repeat(2, auto);
    grid-gap: 0.3rem;
    justify-content: space-between;
    align-items: start;
`


interface SelectFieldProps<Option extends DefaultOption=DefaultOption> {
    onChange: (value: Option | null, event: ActionMeta<Option>) => void;
    name: string;
    label?: string;
    style?: React.CSSProperties;
    defaultValue?: Option;
    placeholder?: Option;
    value?: Option | null;
    options: readonly Option[];
}


export let  SelectField = <Option extends DefaultOption = DefaultOption>(
    props: SelectFieldProps<Option>
): JSX.Element => {


    const style = useMemo(()=>{
        return {
            control:(provided: any,  { data, isDisabled, isFocused, isSelected }: any) => ({
                ...provided,
                minWidth:"7em",
                backgroundColor: ColorTheme.background.select_background,
            }),

            menu: (provided: any,  { data, isDisabled, isFocused, isSelected }: any) => ({
                ...provided,
                width: "max-content",
                backgroundColor: ColorTheme.background.select_background
            }),

            menuPortal: (provided: any,  { data, isDisabled, isFocused, isSelected }: any) => ({
                ...provided,
            }),

            option: (provided: any,  { data, isDisabled, isFocused, isSelected }: any) => ({
                ...provided,
                backgroundColor: isFocused?ColorTheme.background.transparent_accents:undefined
            }),

            singleValue:(provided: any,  { data, isDisabled, isFocused, isSelected }: any) => ({
                ...provided,
                color: ColorTheme.text
            })
        };
    }, [ColorTheme])

    const onSelectChange = useCallback((value: any, event: ActionMeta<any>) => props.onChange(value, event), [props.onChange]);
    let selected_value = props.value ? props.value: null
    selected_value = selected_value ? selected_value : null;

    return(
        <SelectWrapper>
            <Label htmlFor={props.name} >{props.label}</Label>
            <Select
                name={props.name}
                id={props.name}
                styles={style}
                defaultValue={props.defaultValue}
                value={selected_value}
                menuPortalTarget={document.body}
                onChange={onSelectChange}
                options={props.options}
            />
        </SelectWrapper>
    );
};

