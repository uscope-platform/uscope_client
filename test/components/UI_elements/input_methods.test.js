// Copyright 2021 Filippo Savi
// Author: Filippo Savi <filssavi@gmail.com>
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import {ThemeProvider} from "styled-components";

import {
    Button,
    ColorTheme,
    MultiSelect,
    StyledScrollbar,
    Checkbox
} from "../../../src/components/UI_elements";
import {SelectField} from "../../../src/components/UI_elements/Select";
import {fireEvent, render, screen} from "@testing-library/react";
import userEvent from "@testing-library/user-event";

test('button', () => {
    const tree = renderer.create(
        <ThemeProvider theme={ColorTheme}>
            <Button />
        </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})

test('multi select', () => {

    const tree = renderer.create(
        <ThemeProvider theme={ColorTheme}>
            <MultiSelect />
        </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()

})


test('scrollbar', () => {

    const tree = renderer.create(
        <ThemeProvider theme={ColorTheme}>
            <StyledScrollbar/>
        </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()

})

test('simple select', () => {

    let onChangeFired = false;
    let onChange = () => {
        onChangeFired =true
    };


    render(
        <ThemeProvider theme={ColorTheme}>
            <SelectField label="Program type" onChange={onChange} defaultValue={"opt_1"}
                         name="program_type" placeholder="opt_1" options={["opt_1", "opt_2", "opt_3"]}/>
        </ThemeProvider>
    )

    let field = screen.getByDisplayValue("opt_1")

    fireEvent.change(field, {target: {value: 'opt_2'}});
    expect(onChangeFired).toBeTruthy();
    userEvent.selectOptions(
        field,
        screen.getByRole('option', {name: 'opt_2'}),
    )
    expect(screen.getByRole('option', {name: 'opt_2'}).selected).toBe(true)
    expect(onChangeFired).toBeTruthy();
})


test('simple select', () => {

    let onChangeFired = false;
    let onChange = () => {
        onChangeFired =true
    };


    const { container } = render(
        <ThemeProvider theme={ColorTheme}>
            <Checkbox onChange={onChange} name="test_checkbox" value={onChangeFired}/>
        </ThemeProvider>
    )
    const field = container.querySelector(`input[name="test_checkbox"]`);

    fireEvent.click(field);
    expect(onChangeFired).toBeTruthy();

})
