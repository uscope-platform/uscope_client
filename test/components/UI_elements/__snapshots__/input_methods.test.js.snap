// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button 1`] = `
.c0 {
  background: #c76863;
  border-color: #c76863;
  font-weight: bold;
  border-radius: 3px;
  border-style: solid;
  font-size: 16px;
  color: black;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 0.5em 1em;
}

<button
  className="c0"
/>
`;

exports[`multi select 1`] = `
.c1 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-family: Roboto,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-right: 0.5rem;
}

.c0 {
  margin: 0 0.2rem;
  display: grid;
  grid-template-columns: 1fr;
  grid-auto-rows: auto;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: wrap;
  -ms-flex-flow: wrap;
  flex-flow: wrap;
}

<div
  className="c0"
>
  <label
    className="c1"
  />
  <div
    className=" css-b62m3t-container"
    onKeyDown={[Function]}
  >
    <span
      className="css-1f43avz-a11yText-A11yText"
      id="react-select-2-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      className="css-1f43avz-a11yText-A11yText"
    />
    <div
      className=" css-1s2u09g-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className=" css-319lph-ValueContainer"
      >
        <div
          className=" css-14el2xx-placeholder"
          id="react-select-2-placeholder"
        >
          Select...
        </div>
        <div
          className=" css-6j8wv5-Input"
          data-value=""
        >
          <input
            aria-autocomplete="list"
            aria-describedby="react-select-2-placeholder"
            aria-expanded={false}
            aria-haspopup={true}
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect="off"
            className=""
            disabled={false}
            id="react-select-2-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            role="combobox"
            spellCheck="false"
            style={
              Object {
                "background": 0,
                "border": 0,
                "color": "inherit",
                "font": "inherit",
                "gridArea": "1 / 2",
                "label": "input",
                "margin": 0,
                "minWidth": "2px",
                "opacity": 1,
                "outline": 0,
                "padding": 0,
                "width": "100%",
              }
            }
            tabIndex={0}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className=" css-1hb7zxy-IndicatorsContainer"
      >
        <span
          className=" css-1okebmr-indicatorSeparator"
        />
        <div
          aria-hidden="true"
          className=" css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-tj5bde-Svg"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`scrollbar 1`] = `
<div
  style={
    Object {
      "height": "auto",
      "maxHeight": 400,
      "minHeight": 100,
      "overflow": "hidden",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "bottom": undefined,
        "left": undefined,
        "marginBottom": 0,
        "marginRight": 0,
        "maxHeight": 400,
        "minHeight": 100,
        "overflow": "scroll",
        "position": "relative",
        "right": undefined,
        "top": undefined,
      }
    }
  />
  <div
    style={
      Object {
        "borderRadius": 3,
        "bottom": 2,
        "display": "none",
        "height": 6,
        "left": 2,
        "position": "absolute",
        "right": 2,
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "#122964",
          "display": "block",
          "height": "100%",
          "position": "relative",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "borderRadius": 3,
        "bottom": 2,
        "display": "none",
        "position": "absolute",
        "right": 2,
        "top": 2,
        "width": 6,
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "#122964",
          "display": "block",
          "position": "relative",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
